### ISSUE TITLE
Cisco compatible SCP

### ISSUE THREAD

**Issue by bazauto** [sshnet/SSH.NET#108]
I've been working with the Posh-SSH PowerShell module, which uses this library, to transfer files to/from Cisco devices via SCP. With the Cisco devices as the SCP server. Currently testing with a 2801 router running IOS 12.4(24)T1.

We also have newer routers running 15.4 but I have not had chance to test one of those yet.

I've found that the Cisco scp process interprets the quotes added around the filename as part of the filename. If I remove the quotes the file transfer works as expected.

I changed...
```
channel.SendExecRequest(string.Format("scp -t {0}", pathOnly));
```
to ...
```
channel.SendExecRequest(string.Format("scp -t \"{0}\"", pathOnly));
```

It is also worth noting, I'm using the stream based overrides as the Cisco scp process doesn't support the "-p, -r or -d" options.

Is this something you would consider adding compatibility for? Would you like me to propose a solution in a pull request?

**Comment by cocowalla** [comment]
I've had the same problem with Cisco devices - would be nice to get a fix merged in.

**Comment by drieseng** [comment]
@cocowalla @bazauto 
Do you guys have a device that I can test a fix against?

**Comment by darkoperator** [comment]
Probably try with GNS3 if a device is not available, I will check my storage unit I may have an 800 there

**Comment by bazauto** [comment]
I can setup a publicly accessible Cisco device, but it will be next week, if that is any use.

**Comment by drieseng** [comment]
@bazauto Yes, that would definitely be useful. Thanks!

**Comment by drieseng** [comment]
In PR #290 I've introduced a mechanism to disable path quoting, and at the same time we now allow for a custom quoting implementation to be plugged in.

That PR (and the develop branch) also features a new default quoting implementation.
More info on this implementation is available [here](https://github.com/sshnet/SSH.NET/blob/7dde512a9ca662d245b90ad87f28d5a981de637f/src/Renci.SshNet/RemotePathTransformation.cs#L12).

Can you build **SSH.NET** from the **[scp-quoting](https://github.com/sshnet/SSH.NET/tree/scp-quoting)** branch and:
* Try using the new default quoting implementation to see if that works for **CISCO** routers.
* If not, try using the **None** transformation to disable quoting altogether.

Please perform tests with both "regular" paths, and paths with spaces and special characters (#\<\>;%?'" ...).

**Comment by bazauto** [comment]
I've setup a Cisco 2801 router that can be used to test against. It has only a very basic config with a public IP. There is a test file that can be downloaded that is only 10KB.

I've tested from a Linux server using the following command...
-  scp test-file.bin scptest@193.219.111.26:test-file.bin
-  scp scptest@193.219.111.26:test-file.bin test-file.bin

The password for this test user is j8XLZBqm

I've not had chance to test the scp-quoting branch yet, I will see if I get time this week.

**Comment by cocowalla** [comment]
@bazauto best not to post credentials publically! You should probably delete your comment and send to @drieseng at [gertdriesen@msn.com](mailto:gertdriesen@msn.com)!

**Comment by drieseng** [comment]
@bazauto I removed your comment. You may want to modify the password, and send it to me privately.

**Comment by bazauto** [comment]
@drieseng I've changed the password and sent you the new details via e-mail.

Clearly not thinking straight this morning.

**Comment by drieseng** [comment]
With the **None** remote path transformation that landed into the **develop** branch as part of PR #290, I was able to upload and download files that do not contain any special characters.

As I mentioned in my email, I did not find a way to download or upload files from a path containing spaces. The Cisco router does not appear to support any quoting mechanism.

Please reopen this issue if you have more information.

**Comment by drieseng** [comment]
I've just released version **[2016.1.0-beta3](https://github.com/sshnet/SSH.NET/releases/tag/2016.1.0-beta3)** of **SSH.NET**, which includes a fix for this issue.
Please use either the source, binary or **[NuGet](https://www.nuget.org/packages/SSH.NET/2016.1.0-beta3)** distribution to confirm this version resolves the reported problem for you.
